avocado/utils/filelock.py

```diff
@@ -0,0 +1,106 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: Red Hat Inc. 2016
+# Author: Amador Pahim <apahim@redhat.com>
+
+"""
+Utility for individual file access control.
+"""
+
+import os
+import time
+
+from .process import pid_exists
+
+
+class AlreadyLocked(Exception):
+    pass
+
+
+class LockFailed(Exception):
+    pass
+
+
+class FileLock(object):
+
+    """
+    Creates a lock for a file. If the lock file already exists and it
```

Comment:
It may be a good to specify what kind of lock this implements.  What I mean is whether the lock is:
* exclusive
* read-only
* write-only

And of course this is advisory/cooperative lock, so it should be stated.